### PR TITLE
ESLint-based improvements, resurrect schedule links

### PR DESCRIPTION
**Pull request by copperwater**
This adds a number of primarily cosmetic changes to the Javascript code based on ESLint reports, such as changing `==` to `===`. Also the ESLint config file, eslint.json, is added to the config folder.

Also included in this PR is a fixed version of schedule links, which were removed in #195.

### CODE REVIEW

**Comment by Bad-Science** [comment]
@copperwater What was causing the bug described in #195 ?

**Comment by copperwater** [comment]
I believe it was the line `Yacs.observe('selection', scheduleElement, updateSchedules);`. updateSchedules is supposed to take either an array of section IDs to call the schedule API with, or nothing/undefined, in which case it will use the current selections from the cookie.

Because the observer uses addEventListener to attach the callback function, and the raw updateSchedules was being used as the callback, updateSchedules was indeed being called whenever selections changed. But using it as an event listener meant that the first argument passed to it was an event object. Since that isn't an array of ints or undefined, it didn't know what to do with it, so it did nothing.

The fix was to make the event listener callback an anonymous function that calls updateSchedules.

File: app/assets/javascripts/controllers/schedules/controller.js [hunk 1]
```diff
@@ -146,11 +204,34 @@ Yacs.views.schedules = function (target, params) {
...
       scheduleStatusElement.textContent = scheduleStatusStr;
     }
+  }
+
+  /*
+   * Generate a link to this set of schedules from current selections
+   * and copy it to the user's clipboard.
+   */
+  var copyScheduleLink = function() {
+    console.log('sdfsdf');
```
Comment (Bad-Science) [inline app/assets/javascripts/controllers/schedules/controller.js]: remove this pls :)

Comment (copperwater) [inline app/assets/javascripts/controllers/schedules/controller.js]: fixed

**Pull request merged**